### PR TITLE
Convert data model

### PR DESCRIPTION
**Pull request by MarkPaxton**

### CODE REVIEW

**Comment by MarkPaxton** [comment]
This incorporates PR #1 and I have changed the namespace to uk.gov.hmrc.ssttp.

File: app/models/EligibilityDebtType.scala [hunk 1]
```diff
@@ -14,15 +14,12 @@
...
-  */
-case class LiabilityResult(value: String)
+case class EligibilityDebtType(hasSelfAssessmentDebt: Boolean, hasOtherDebt: Boolean)
 
-  object LiabilityResult {
-    implicit val formats = Json.format[LiabilityResult]
-  }
+object EligibilityDebtType {
+  implicit val format = Json.format[EligibilityDebtType]
+}
```
Comment (nedda76) [inline app/models/EligibilityDebtType.scala]: Our flow for non-logged in journey no longer includes eligibility  - so the debtType and existingTtp models can be removed.

File: app/models/EligibilityExistingTTP.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+package models
+
+import play.api.libs.json.Json
+
+case class EligibilityExistingTTP(hasExistingTTP: Option[Boolean])
+
+object EligibilityExistingTTP {
+  implicit val format = Json.format[EligibilityExistingTTP]
+}
+
```
Comment (nedda76) [inline app/models/EligibilityExistingTTP.scala]: same as above - can be removed, no longer part of the flow.

File: conf/app.routes [hunk 3]
```diff
@@ -5,11 +5,11 @@
...
-#GET    /eligibility                     controllers.EligibilityController.present()
-#GET    /eligibility/debt-type           controllers.EligibilityController.debtTypePresent()
-#POST   /eligibility/debt-type           controllers.EligibilityController.debtTypeSubmit()
-#GET    /eligibility/existing-ttp        controllers.EligibilityController.existingTtpPresent()
-#POST   /eligibility/existing-ttp        controllers.EligibilityController.existingTtpSubmit()
+GET    /eligibility                     controllers.EligibilityController.present()
+GET    /eligibility/debt-type           controllers.EligibilityController.debtTypePresent()
+POST   /eligibility/debt-type           controllers.EligibilityController.debtTypeSubmit()
+GET    /eligibility/existing-ttp        controllers.EligibilityController.existingTtpPresent()
+POST   /eligibility/existing-ttp        controllers.EligibilityController.existingTtpSubmit()
```
Comment (nedda76) [inline conf/app.routes]: All the eligibility routes, models, controllers and pages should be removed as they are no longer part of the non-logged in journey flow.

File: test/models/EligibilityExistingTTP.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      ettp should not be (null)
+      ettp.hasExistingTTP.get should be (false)
+    }
+
+    "create with false no option parameter" in {
+      val ettp = new EligibilityExistingTTP(None)
+      ettp should not be (null)
+      ettp.hasExistingTTP.isEmpty should be (true)
+    }
+}
```
Comment (nedda76) [inline test/models/EligibilityExistingTTP.scala]: The specs for these can be removed as well - eligibility related.

**Review by nedda76 — CHANGES_REQUESTED** [review]

**Pull request merged**